cmd/zq/zq.go

```diff
@@ -249,16 +249,22 @@ func (c *Command) Run(args []string) error {
 	if err != nil {
 		return err
 	}
-	defer writer.Close()
 	mux, err := driver.CompileWarningsCh(context.Background(), query, reader, false, nano.MaxSpan, zap.NewNop(), wch)
 	if err != nil {
+		writer.Close()
 		return err
 	}
 	d := driver.NewCLI(writer)
 	if !c.quiet {
 		d.SetWarningsWriter(os.Stderr)
 	}
-	return driver.Run(mux, d, nil)
+	if err := driver.Run(mux, d, nil); err != nil {
+		return err
+	}
+	// If writer is an s3 error it could have important information as to
+	// whether the request was written successfully. Cannot do a simple defer
+	// writer.Close() here.
```

Comment:
I think you can leave out the comment: for any writecloser, it's good practice to not discard the close error. @nwt called this recently on a separate PR of mine, and I think took the code from here; I should have come back to update this as well.

Reply:
Sounds good.

Reply:
Done w/ pr description updated.